unittests/AST/ASTImporterTest.cpp

```diff
@@ -907,18 +908,63 @@ TEST(ImportDecl, DISABLED_ImportTemplateDefaultArgument) {
           functionTemplateDecl(has(templateArgument()))));
 }
 
-TEST_F(Fixture, ImportTemplatedDecl) {
+TEST_F(Fixture, ImportOfTemplatedDeclOfClassTemplateDecl) {
   Decl *FromTU = getTuDecl("template<class X> struct S{};", Lang_CXX);
   auto From =
       FirstDeclMatcher<ClassTemplateDecl>().match(FromTU, classTemplateDecl());
   ASSERT_TRUE(From);
   auto To = cast<ClassTemplateDecl>(Import(From, Lang_CXX));
   ASSERT_TRUE(To);
-  auto ToTemplated = To->getTemplatedDecl();
-  auto ToTemplated1 =
-      cast<CXXRecordDecl>(Import(From->getTemplatedDecl(), Lang_CXX));
+  Decl *ToTemplated = To->getTemplatedDecl();
+  Decl *ToTemplated1 = Import(From->getTemplatedDecl(), Lang_CXX);
   EXPECT_TRUE(ToTemplated1);
-  ASSERT_EQ(ToTemplated1, ToTemplated);
+  EXPECT_EQ(ToTemplated1, ToTemplated);
+}
+
+// Currently disabled, because lookup fails to find the existing templated decl.
+TEST_F(Fixture, DISABLED_ImportOfTemplatedDeclOfFunctionTemplateDecl) {
```

Comment:
Note, this fails in the baseline too, there is a flaw in the current lookup logic, which is orthogonal to the changes of this PR.

Reply:
Probably this is what I try to solve right now. (Or the other direction: import of `FunctionDecl` that is a TemplatedDecl of a `FunctionTemplateDecl`.)